lib/import/colorectal/providers/sheffield/sheffield_handler_colorectal.rb

```diff
@@ -7,618 +7,413 @@ module Providers
-                genocolorectal.add_test_scope(:targeted_mutation)
-                @logger.debug "ADDED TARGETED TEST for: #{record.raw_fields['karyotypingmethod']}"
-              else
-                genocolorectal.add_test_scope(:full_screen)
-                @logger.debug "ADDED FULL_SCREEN TEST for: #{record.raw_fields['karyotypingmethod']}"
-              end
-            elsif geno == 'R210 :: Inherited MMR deficiency (Lynch syndrome)'
-              if karyo == 'R242.1 :: Predictive testing' || karyo == 'R242.1 :: Predictive MLPA'
-                genocolorectal.add_test_scope(:targeted_mutation)
-                @logger.debug "ADDED TARGETED TEST for: #{record.raw_fields['karyotypingmethod']}"
-              else
-                genocolorectal.add_test_scope(:full_screen)
-                @logger.debug "ADDED FULL_SCREEN TEST for: #{record.raw_fields['karyotypingmethod']}"
-              end
-            elsif geno == 'R209 :: Inherited colorectal cancer (with or without polyposis)'
-              if karyo == 'R242.1 :: Predictive testing'
-                genocolorectal.add_test_scope(:targeted_mutation)
-                @logger.debug "ADDED TARGETED TEST for: #{record.raw_fields['karyotypingmethod']}"
-              else
-                genocolorectal.add_test_scope(:full_screen)
-                @logger.debug "ADDED FULL_SCREEN TEST for: #{record.raw_fields['karyotypingmethod']}"
-              end
-            elsif geno == 'R211 :: Inherited polyposis - germline test'
-              if karyo == 'R211.1 :: APC and MUTYH genes in Leeds'
-                genocolorectal.add_test_scope(:full_screen)
-                @logger.debug "ADDED FULL_SCREEN for: #{record.raw_fields['karyotypingmethod']}"
-              elsif karyo == 'R387.1 ::  APC and MUTYH analysis only'
-                genocolorectal.add_test_scope(:full_screen)
-                @logger.debug "ADDED FULL_SCREEN for: #{record.raw_fields['karyotypingmethod']}"
-              else
-                genocolorectal.add_test_scope(:targeted_mutation)
-                @logger.debug "ADDED TARGETED TEST for: #{record.raw_fields['karyotypingmethod']}"
-              end
-            elsif geno == 'Breast Ovarian & Colorectal cancer panel'
-              if karyo == 'Full panel'
-                genocolorectal.add_test_scope(:full_screen)
-                @logger.debug "ADDED FULL_SCREEN for: #{record.raw_fields['karyotypingmethod']}"
-              end
+          def add_test_scope_from_geno_karyo(genocolorectal, record)
+            genotype_str = record.raw_fields['genetictestscope'].strip
+            karyo = record.raw_fields['karyotypingmethod'].strip
+            moleculartestingtype = record.raw_fields['moleculartestingtype'].strip
+            process_method = GENETICTESTSCOPE_METHOD_MAPPING[genotype_str.downcase]
+            if process_method
+              send(process_method, karyo, genocolorectal, moleculartestingtype)
```

Comment:
[`public_send`](https://apidock.com/ruby/Object/public_send) should work here, I don't think any of the methods are private or protected?

Reply:
yes , changed to public_send